packages/ruleset/src/functions/check-major-version.js

```diff
@@ -88,9 +106,44 @@ module.exports = targetVal => {
   }
 };
 
-// Return the first segment of a path that matches the pattern 'v\d+'
-function getVersion(path) {
-  const url = new URL(path, 'https://foo.bar');
+/**
+ * Returns the first segment of the path portion of "urlString" that matches the pattern 'v\d+'
+ * @param {*} urlString an absolute or relative URL string (e.g. "https://myhost.com/api/v1" or "/v1/clouds").
+ * @returns the first path segment that appears to indicate the API major version (e.g. "v1").
+ */
+function getVersion(urlString) {
+  const url = new URL(urlString, 'https://foo.bar');
   const segments = url.pathname.split('/');
   return segments.find(segment => segment.match(/v[0-9]+/));
 }
+
+/**
+ * Returns the "default" server URL for the specified server object.
+ *
+ * @param {object} server an entry from the API definition's "servers" list.
+ * @returns the default URL string which is the result of replacing server
+ * variable references found in the server object's "url" field with
+ * the default value for each referenced server variable.
+ */
+function getDefaultUrl(server) {
+  let urlString = null;
+  if (server && server.url) {
+    urlString = server.url;
+
+    if (server.variables) {
+      for (const variable of Object.entries(server.variables)) {
+        const varName = variable[0];
+        const varObj = variable[1];
+        if (varName && varObj) {
+          const varRefRE = new RegExp('{' + variable[0] + '}', 'g');
+          const value = varObj.default;
+          if (varObj.default) {
```

Comment:
This is minor and doesn't really matter but since you've already set the variable, this would be equivalent

```suggestion
          if (value) {
```